utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,219 @@
+    test_hists.extend(glob.glob("%s/%s.%s*.h.nc%s" % (from_dir, testcase, model, suffix)))
+
+    # Match multi-instance files produced by run
+    test_hists.extend(glob.glob("%s/%s.%s*.h?.*.nc%s" % (from_dir, testcase, model, suffix)))
+    test_hists.extend(glob.glob("%s/%s.%s*.h.*.nc%s" % (from_dir, testcase, model, suffix)))
+
+    # suffix == "" implies baseline comparison, baseline hist files have simpler names
+    if suffix == "":
+        test_hists.extend(glob.glob("%s/%s.h.nc" % (from_dir, model)))
+        test_hists.extend(glob.glob("%s/%s.h?.nc" % (from_dir, model)))
+
+    test_hists.sort()
+    return test_hists
+
+def move(case, suffix):
+    """
+    Change the suffix for the most recent batch of hist files in a case. This can
+    allow you to temporarily "save" these files so they won't be blown away if you
+    re-run the case.
+
+    case - The case containing the files you want to save
+    suffix - The string suffix you want to add to saved files, this can be used to find them later.
+    """
+    rundir   = case.get_value("RUNDIR")
+    testcase = case.get_value("CASE")
+
+    # Loop over models
+    comments = "Moving hist files to suffix '%s'\n" % suffix
+    num_moved = 0
+    for model in _iter_model_file_substrs(case):
+        comments += "  Moving hist files for model '%s'\n" % model
+        test_hists = _get_all_hist_files(testcase, model, rundir)
+        num_moved += len(test_hists)
+        for test_hist in test_hists:
+            new_file = "%s.%s" % (test_hist, suffix)
+            if os.path.exists(new_file):
+                os.remove(new_file)
+
+            comments += "    Copying '%s' to '%s'\n" % (test_hist, new_file)
+            shutil.copy(test_hist, new_file)
+
+    expect(num_moved > 0, "move failed: no hist files found in rundir '%s'" % rundir)
+
+    return comments
+
+def _compare_hists(case, from_dir1, from_dir2, suffix1="", suffix2=""):
+    if from_dir1 == from_dir2:
+        expect(suffix1 != suffix2, "Comparing files to themselves?")
+
+    testcase = case.get_value("CASE")
+    all_success = True
+    num_compared = 0
+    comments = "Comparing hists for case '%s' dir1='%s', suffix1='%s',  dir2='%s' suffix2='%s'\n" % \
+        (testcase, from_dir1, suffix1, from_dir2, suffix2)
+
+    for model in _iter_model_file_substrs(case):
+        comments += "  comparing model '%s'\n" % model
+        hists1 = _get_all_hist_files(testcase, model, from_dir1, suffix1)
+        hists2 = _get_all_hist_files(testcase, model, from_dir2, suffix2)
+        len_hist1 = len(hists1)
+        len_hist2 = len(hists2)
+        if len_hist1 == 0:
+            comments += " no hist files found for model %s\n"%model
+            continue
+        if len_hist1 != len_hist2:
+            comments += "    num hists does not match %d != %d\n" % (len_hist1, len_hist2)
+            all_success = False
+            continue
+
+        num_compared += len(hists1)
+        for hist1, hist2 in zip(hists1, hists2):
```

Comment:
Added a new function _get_latest_hist_files which finds only the most recent history file for each model and extension type.   This really reduces the time required for the test and fixes the issue with matching to the baseline files.